lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvStatement.php

```diff
@@ -197,8 +197,15 @@ public function execute($params = null)
             }
         }
 
-        $this->stmt = sqlsrv_query($this->conn, $this->sql, $this->params);
         if ( ! $this->stmt) {
+            $stmt = sqlsrv_prepare($this->conn, $this->sql, $this->params);
+            if (!$stmt) {
```

Comment:
This branch should be tested.

Also, add empty lines around control structures